feat(docs): main troubleshooting page

charislam opened the pull request:
feat: add troubleshooting pages to docs

preview: https://docs-git-feat-docs-troubleshooting-2-supabase.vercel.app/docs/guides/troubleshooting

not in nav yet until we port a significant amount of these over

charislam committed:
feat(docs): main troubleshooting page

charislam committed:
feat(docs): global troubleshooting page layout

charislam committed:
feat(docs): global troubleshooting page filter logic

charislam committed:
enhance(docs): troubleshooting page ui & filters

Improve UX and accessibility of global troubleshooting search page.

- Implement TroubleshootingFilterStateProvider for managing filter state
- Add TroubleshootingFilterEmptyState component
- Improve TroubleshootingFilter UI with applied filters display
- Update TroubleshootingPreview component with better layout and accessibility
- Add TroubleshootingEntryAssociatedErrors component for displaying related errors
- Refactor Troubleshooting utils and schemas
- Update troubleshooting content template

charislam committed:
feat(docs): individual troubleshooting pages

Add individual pages for detailed information on troubleshooting
entries. Small fixes to UI for troubleshooting displays.

- Wrap GlobalTroubleshootingPage with SidebarSkeleton
- Add TroubleshootingErrorListDetailed component for detailed error display
- Implement TroubleshootingBackLink component for navigation
- Update TroubleshootingPreview to use Next.js Link and include parent page
- Refactor Footer component to accept className prop
- Add hideFooter option to SidebarSkeleton
- Minor updates to TopNavBar and other utility functions

charislam commented on apps/docs/app/guides/(with-sidebar)/[...slug]/page.tsx:
Moved all of these under a nested directory so we could make the troubleshooting pages have a different layout

charislam commented on apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:
```diff
@@ -172,6 +172,11 @@ export const GLOBAL_MENU_ITEMS: GlobalMenuItems = [
       label: 'Resources',
       menuItems: [
         [
+          /*           {
```
will eventually go here but hidden for now because we don't have many migrated yet

charislam commented on apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:
```diff
@@ -172,6 +172,11 @@ export const GLOBAL_MENU_ITEMS: GlobalMenuItems = [
         [
+          /*           {
+            label: 'Troubleshooting',
+            icon: 'contributing',
```
@fsansalvadore any suggestions for icons? currently contributing uses the wrench icon, which probably should be transferred to troubleshooting, but then I'm not sure what to use for contributing

fsansalvadore replied:
`Contributing` could either be a simple _heart_ or a _"branching"_ icon, maybe?

<img width="26" alt="Screenshot 2024-09-11 at 14 24 08" src="https://github.com/user-attachments/assets/c2ecc4e1-ace4-46f5-bae3-3221b3cab02b">

And I agree troubleshooting would be the wrench.

charislam commented on apps/docs/content/troubleshooting/monitor-supavisor-postgres-connections.mdx:
```diff
@@ -0,0 +1,127 @@
+"performance",
+"pooler",
+]
+
+[[errors]]
+http_status_code = 404
+code = "not_found"
+message = "The resource was not found"
+
+[[errors]]
+http_status_code = 403
+code = "forbidden"
+message = "Not allowed to access this resource"
+
+[[errors]]
+http_status_code = 500
+code = "internal_server_error"
+message = "Internal server error"
```
Gonna remove these later as none of them are actually relevant to the issue, just needed some dummy data to see how the display looks

charislam replied:
```suggestion
```

charislam committed:
fix: minor styling issues

charislam commented on apps/docs/app/guides/(with-sidebar)/[...slug]/page.tsx:
moved all of these under a nested directory so we could have a sidebarless layout for troubleshooting

fsansalvadore commented:
@charislam what if the filtering form and topics were in the sidebar? 
It would keep the layout more consistent.

Plus, can we also reuse the breadcrumbs component?

charislam committed:
tweak(troubleshooting docs): use breadcrumbs

charislam committed:
tweak(troubleshooting docs): use sidebar for filter

charislam committed:
enhance(troubleshooting docs): navigation, keywords

Better navigation for individual troubleshooting entries, using a
version of the sidebar + keywords handling via search params for good
linking behavior

Details:

- Add TroubleshootingSidebar to TroubleshootingPage
- Implement query state management for keywords using nuqs
- Improve accessibility and styling of troubleshooting components
- Update empty state handling and related keywords display
- Upgrade nuqs package to version 1.19.1

charislam committed:
fix(docs): wrap useSearchParams in Suspense

charislam commented:
@fsansalvadore what do you think with the sidebar?

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/640a5dc3-860d-4fb8-bb54-721381139583">

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/d92730f8-2e7f-473e-8cc9-96f4004275b8">

charislam committed:
seo(docs): add canonical link to troubleshooting

charislam committed:
enhance(docs,troubleshooting): auto-reveal errors

On troubleshooting search pages, if the searched term matches a hidden
error, auto-reveal it when searching. When the search term is cleared,
hide programmatically-opened errors while preserving user-opened ones.

charislam committed:
enhance(docs,troubleshooting): search bar

Add search bar to individual troubleshooting pages

charislam committed:
fix(docs): breadcrumb links

Fix breadcrumb linking so it uses next/link instead of reloading entire
app on navigation.

charislam committed:
fix(docs): typo

charislam committed:
enh(docs): new troubleshooting design

charislam committed:
enh(docs): use new multi-select

fsansalvadore reviewed (approved):
The mobile version needs a bit of design love, but since this isn't user facing yet we can keep it moving 👍 

we need Dashboard team to approve also, not sure why that `apps/studio/next-env.d.ts` keeps changing...

charislam committed:
enh(docs): troubleshooting responsive styles

charislam committed:
Update apps/docs/content/troubleshooting/monitor-supavisor-postgres-connections.mdx
